wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -216,14 +231,16 @@ public Connection connect(final String url, final Properties info) throws SQLExc
         effectiveConnectionProvider = configurationProfile.getConnectionProvider();
       }
 
+      ServiceContainer serviceContainer = new ServiceContainerImpl(storageService, monitorService, telemetryFactory);
```

Comment:
Can we use a factory approach instead of direct instantiated a class? That's mostly for testing purposes.